Macarena2018/src/main/java/frc/team281/robot/subsystems/LinearAxis.java

```diff
@@ -0,0 +1,164 @@
+
+/**
+ * Simple implementation of a linear axis.
+ * Doesnt use PID-- the assumption is that we can get close
+ * Uses a max speed for 'cruise', and then slows down when it gets near the ends.
+ * enough with simple control
+ * @author dcowden
+ *
+ */
+public class LinearAxis {
+
+    private DataLogger dataLogger;
+    private Encoder axisEncoder;
+    private WPI_TalonSRX talon;
+    private DigitalInput startLimitSwitch;
+    private DigitalInput endLimitSwitch;
+    
+    private int lengthCounts = 0;
+    private int positionTolerance = 0;
+    private int desiredPositionCounts = 0;
+    private double cruiseSpeedPercent = 0.0;
+    private double minSpeedPercent = 0.0;
+    private double rampCounts = 0.0;
+    private double currentSpeed = 0.0;
+    private double inverted = 1.0;
+    private MoveState moveState = MoveState.STOPPED;
+    
+    public enum MoveState {
+        MOVING_FORWARD,
+        MOVING_BACK,
+        STOPPED
+    }
+    
+    public LinearAxis(Encoder encoder, DigitalInput startLimit, DigitalInput endLimit, WPI_TalonSRX talon ,
+            int lengthCounts, double cruiseSpeedPercent, double minSpeedPercent, int rampCounts, boolean inverted, int positionTolerance){
+        dataLogger = DataLoggerFactory.getLoggerFactory().createDataLogger("LinearAxis");
+        this.axisEncoder = encoder;
+        this.startLimitSwitch = startLimit;
+        this.endLimitSwitch = endLimit;
+        this.talon = talon;
+        this.lengthCounts = lengthCounts;
+        this.cruiseSpeedPercent = cruiseSpeedPercent;
+        this.positionTolerance = positionTolerance;
+        this.minSpeedPercent = minSpeedPercent;
+        this.rampCounts = rampCounts;
+        if ( inverted){
+            this.inverted = -1.0;
+        }
+        
+    }
+    
+    public MoveState getMoveState(){
+        return this.moveState;
+    }
+    
+    public void moveToStart(){
+        moveToPosition(0);
+    }
+    public void moveToEnd(){
+        moveToPosition(this.lengthCounts);
+    }
+    
+    public void moveToPosition(int positionCounts){
+        desiredPositionCounts = limitCommand(positionCounts);
+    }
+    
+    public void init(){
+        //assume we're starting on the bottom
+        axisEncoder.reset();
+        talon.set(ControlMode.PercentOutput, 0.0);
+    }
+    
+    public void stop(){
+        talon.set(0.0);
+        currentSpeed = 0.0;
+        moveState = MoveState.STOPPED;
+    }
+    
+    public double getSpeed(){
+        return this.currentSpeed;
+    }
+    
+    public int getCommandedPosition(){
+        return desiredPositionCounts;
+    }
+    
+    public int getPosition(){
+        return axisEncoder.get();
+    }
+    
+    public void periodic(){
+        dataLogger.log("Position",getPosition());
+        dataLogger.log("Speed",getSpeed() );
+        dataLogger.log("State",this.moveState+"");
+        checkLimits();
+        computeSpeed();
+        talon.set(inverted*currentSpeed);
```

Comment:
Can we make all the talon.set calls use the same style.  Both this one and the line 83 use the WPI specific single parameter call.  I'd rather see everything using:
     talon.set(ControlMode.PercentOutput, inverted*curentSpeed);